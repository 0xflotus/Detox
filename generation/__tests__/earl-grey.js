const fs = require("fs");
const remove = require("remove");
const earlGreyGenerator = require("../earl-grey");

describe("earl-grey generation", () => {
  let ExampleClass;
  let exampleContent;
  beforeAll(() => {
    // Generate the code to test
    fs.mkdirSync("./__tests__/generated");

    const files = {
      "./fixtures/example.h": "./__tests__/generated/example.js"
    };

    earlGreyGenerator(files);

    // Load
    ExampleClass = require("./generated/example.js");
    exampleContent = fs.readFileSync(
      "./__tests__/generated/example.js",
      "utf8"
    );
  });

  it("should export the class", () => {
    expect(ExampleClass.actionForMultipleTapsWithCount).toBeInstanceOf(
      Function
    );
  });

  describe("Comments", () => {
    it("should include single line comments", () => {
      expect(exampleContent.indexOf("Single Line Comment here")).not.toBe(-1);
    });

    it("should include multi line comments", () => {
      expect(
        exampleContent.indexOf("Multi Line Comment here\nAwesome")
      ).not.toBe(-1);
    });
  });

  describe("Error handling", () => {
    it("should throw with too few arguments");
    
    it("should throw error for wrong type", () => {
      expect(() => {
        ExampleClass.actionForMultipleTapsWithCount("foo");
      }).toThrowErrorMatchingSnapshot();

      expect(() => {
        ExampleClass.actionForMultipleTapsWithCount(42);
      }).not.toThrow();

      expect(() => {
        ExampleClass.actionForMultipleTapsWithCountatPoint(3, 4);
      }).toThrowErrorMatchingSnapshot();

      expect(() => {
        ExampleClass.actionForMultipleTapsWithCountatPoint(42, { x: 1, y: 2 });
      }).not.toThrow();
    });

    it("should throw error for not in accepted range", () => {
      expect(() => {
        ExampleClass.actionForScrollInDirectionamountxOriginStartPercentageyOriginStartPercentage(
          "flipside",
          3,
          4,
          5
        );
      }).toThrowErrorMatchingSnapshot();

      expect(() => {
        ExampleClass.actionForScrollInDirectionamountxOriginStartPercentageyOriginStartPercentage(
          "down",
          3,
          4,
          5
        );
      });
    });
  });

  afterAll(() => {
    // Clean up
    remove.removeSync("./__tests__/generated");
  });
});
