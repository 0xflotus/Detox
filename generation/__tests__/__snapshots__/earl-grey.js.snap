// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`earl-grey generation Error handling should throw error for not in accepted range 1`] = `"direction should be one of [Left, Right, Up, Down], but got flipside"`;

exports[`earl-grey generation Error handling should throw error for wrong type 1`] = `"count should be a number, but got foo"`;

exports[`earl-grey generation Error handling should throw error for wrong type 2`] = `"point should be a object, but got 4"`;
